src/pyfmi/fmi_util.pyx

```diff
@@ -392,46 +420,53 @@ cpdef convert_sorted_vars_name_desc(list sorted_vars):
         FMIL.memcpy(&name_output[kn], ctmp_name, name_length_trial)
         FMIL.memcpy(&desc_output[kd], ctmp_desc, desc_length_trial)
 
-    py_desc_string = desc_output[:(items+1)*desc_length]
-    py_name_string = name_output[:(items+1)*name_length]
+    py_desc_string = desc_output[:(items+nof_diag_params+nof_diag_vars+1)*desc_length]
+    py_name_string = name_output[:(items+nof_diag_params+nof_diag_vars+1)*name_length]
 
     FMIL.free(name_output)
     FMIL.free(desc_output)
 
     return name_length, py_name_string, desc_length, py_desc_string
 
-cpdef convert_sorted_vars_name(list sorted_vars):
+cpdef convert_sorted_vars_name(list sorted_vars, list diag_param_names, list diag_vars):
     cdef int items = len(sorted_vars)
+    cdef int nof_diag_params = len(diag_param_names)
+    cdef int nof_diag_vars = len(diag_vars)
     cdef int i, name_length_trial, kn
     cdef list name = [encode("time")]
     cdef int name_length = len(name[0])+1
     cdef char *name_output
     cdef char *ctmp_name
 
-    for i in range(items):
-        var = sorted_vars[i]
-        tmp_name = encode(var.name)
+    for i in range(items+nof_diag_params+nof_diag_vars):
+        if i < items:
+            var = sorted_vars[i]
+            tmp_name = encode(var.name)
+        elif i < items+nof_diag_params:
+            tmp_name = encode(diag_param_names[i-items][0])
+        else:
+            tmp_name = encode(diag_vars[i-items-nof_diag_params][0])
```

Comment:
This could also use a generator expression like the one above.

Reply:
Fixed